### ISSUE TITLE
[CI] DriverTests testThreadContext failing

### ISSUE THREAD

**Issue by mark-vieira** [elastic/elasticsearch#101095]
This didn't reproduce for me but it's a newly added test and has already failed so is probably worth a look.

**Build scan:**
https://gradle-enterprise.elastic.co/s/jnsn6r7agwy3s/tests/:x-pack:plugin:esql:compute:test/org.elasticsearch.compute.operator.DriverTests/testThreadContext
**Reproduction line:**
```
./gradlew ':x-pack:plugin:esql:compute:test' --tests "org.elasticsearch.compute.operator.DriverTests.testThreadContext" -Dtests.seed=F277C2E26160F066 -Dtests.locale=ar-SA -Dtests.timezone=Australia/Currie -Druntime.java=21
```

**Applicable branches:**
main

**Reproduces locally?:**
No

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.compute.operator.DriverTests&tests.test=testThreadContext
**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: "user1"
     but: was null

  at __randomizedtesting.SeedInfo.seed([F277C2E26160F066:6474F7E2FC8F4A3A]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.compute.operator.DriverTests.assertRunningWithRegularUser(DriverTests.java:89)
  at org.elasticsearch.compute.operator.DriverTests.lambda$testThreadContext$2(DriverTests.java:67)
  at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:128)
  at org.elasticsearch.action.support.SubscribableListener$SuccessResult.complete(SubscribableListener.java:310)
  at org.elasticsearch.action.support.SubscribableListener.tryComplete(SubscribableListener.java:230)
  at org.elasticsearch.action.support.SubscribableListener.addListener(SubscribableListener.java:133)
  at org.elasticsearch.action.support.SubscribableListener.addListener(SubscribableListener.java:108)
  at org.elasticsearch.compute.operator.DriverTests.testThreadContext(DriverTests.java:66)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  at java.lang.reflect.Method.invoke(Method.java:580)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1583)

```

**Comment by mark-vieira** [comment]
@dnhatn do you mind taking a look at this one?

**Comment by dnhatn** [comment]
@mark-vieira Thanks for ping. I am taking a look now.

**Comment by dnhatn** [comment]
I've removed the blocker label as I believe this is a test bug.

**Comment by dnhatn** [comment]
I've opened https://github.com/elastic/elasticsearch/pull/101097.

**Comment by danielmitterdorfer** [comment]
I have to reopen this as this has failed again on `main`:

* [Build scan](https://gradle-enterprise.elastic.co/s/3q7b5t6cgrsma/tests/task/:x-pack:plugin:esql:compute:test/details/org.elasticsearch.compute.operator.DriverTests/testThreadContext?expanded-stacktrace=WyIwIl0&top-execution=1)

Build output:

```
java.lang.AssertionError: 	
Expected: "user1"	
     but: was "system"	
	at __randomizedtesting.SeedInfo.seed([24B86A2CAD86ADC2]:0)	
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)	
	at org.junit.Assert.assertThat(Assert.java:956)	
	at org.junit.Assert.assertThat(Assert.java:923)	
	at org.elasticsearch.compute.operator.DriverTests.assertRunningWithRegularUser(DriverTests.java:90)	
	at org.elasticsearch.compute.operator.DriverTests.lambda$testThreadContext$2(DriverTests.java:66)	
	at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:128)	
	at org.elasticsearch.action.support.SubscribableListener$SuccessResult.complete(SubscribableListener.java:310)	
	at org.elasticsearch.action.support.SubscribableListener.tryComplete(SubscribableListener.java:230)	
	at org.elasticsearch.action.support.SubscribableListener.setResult(SubscribableListener.java:259)	
	at org.elasticsearch.action.support.SubscribableListener.onResponse(SubscribableListener.java:173)	
	at org.elasticsearch.action.ActionListener$4.onResponse(ActionListener.java:331)	
	at org.elasticsearch.action.support.RefCountingListener.finish(RefCountingListener.java:121)	
	at org.elasticsearch.core.AbstractRefCounted$1.closeInternal(AbstractRefCounted.java:117)	
	at org.elasticsearch.core.AbstractRefCounted.decRef(AbstractRefCounted.java:69)	
	at org.elasticsearch.action.support.RefCountingRunnable.close(RefCountingRunnable.java:116)	
	at org.elasticsearch.core.Releasables$3.close(Releasables.java:142)	
	at org.elasticsearch.action.support.RefCountingListener$1.onResponse(RefCountingListener.java:151)	
	at org.elasticsearch.action.support.RefCountingListener$1.onResponse(RefCountingListener.java:146)	
	at org.elasticsearch.action.support.SubscribableListener$SuccessResult.complete(SubscribableListener.java:310)	
	at org.elasticsearch.action.support.SubscribableListener.tryComplete(SubscribableListener.java:230)	
	at org.elasticsearch.action.support.SubscribableListener.setResult(SubscribableListener.java:259)	
	at org.elasticsearch.action.support.SubscribableListener.onResponse(SubscribableListener.java:173)	
	at org.elasticsearch.compute.operator.DriverContext$AsyncActions.removeInstance(DriverContext.java:176)	
	at org.elasticsearch.compute.operator.DriverContext.removeAsyncAction(DriverContext.java:159)	
	at org.elasticsearch.action.ActionListenerImplementations$RunAfterActionListener.onResponse(ActionListenerImplementations.java:262)	
	at org.elasticsearch.action.ActionListener$4.onResponse(ActionListener.java:331)	
	at org.elasticsearch.compute.operator.DriverTests$SwitchContextOperator.lambda$performAsync$0(DriverTests.java:124)	
	at org.elasticsearch.action.ActionRunnable$3.doRun(ActionRunnable.java:73)	
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:983)	
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)	
	at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)	
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:983)	
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)	
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)	
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)	
	at java.lang.Thread.run(Thread.java:1583)
```

Reproduction line: `./gradlew ':x-pack:plugin:esql:compute:test' --tests "org.elasticsearch.compute.operator.DriverTests.testThreadContext" -Dtests.seed=24B86A2CAD86ADC2 -Dtests.locale=ar-KW -Dtests.timezone=Europe/Brussels -Druntime.java=21`

Reproduces locally: no

**Comment by dnhatn** [comment]
The test failed because of async actions introduced in #102508. I have opened the fix in #102550.

**Comment by DaveCTurner** [comment]
I've muted this test in 0ecb2af13d9910f01f077f02a95c5eb3e5dfab53 since this is blocking other work while the fix is awaiting review. You'll need to unmute it in #102550.
